components/discord_webhook/discord_webhook.app.js

```diff
@@ -62,5 +66,33 @@ module.exports = {
       }
       return resp.data;
     },
+    async sendMessageWithFile({
+      content, username, avatarURL, threadID, file,
+    }) {
+      const data = new FormData();
+      if (content) data.append("content", content);
+      if (username) data.append("username", username);
+      if (avatarURL) data.append("avatar_url", avatarURL);
+      if (file) data.append("file", file);
+      const resp = await axios({
+        method: "POST",
+        url: this.url(),
+        headers: {
+          "Content-Type": "multipart/form-data; boundary=" + data._boundary,
+        },
+        validateStatus: () => true,
+        params: {
+          thread_id: threadID
```

Comment:
I would do just:

```javascript
params: {
  thread_id: threadID,
}
```

I see that `sendMessage` does the same as you did here, so if you don't mind, please refactor that method, as well.

Reply:
Updated these and added `if (!threadID) threadID = undefined;` above to avoid getting this error:
`Error: {"thread_id":["Value \"\" is not snowflake."]}`